### PR TITLE
chore: Migrate issue templates to Github forms

### PR DESCRIPTION
**Pull request by jszwedko**
This:

* Drops the enhancement template. I don't think there is any practical
  difference, from a user's perspective, between this and a feature.
* Drops the task template. Only we use this internally and most of the
  time the labels need to be adjusted.
* Migrates the bug and feature templates to use Github's new issue form
  feature. I also reorganized them a bit while migrating.

Ref: https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms

Signed-off-by: Jesse Szwedko <jesse@szwedko.me>

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug.yml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+      including what currently happens and what you expected to happen.
+  validations:
+    required: true
+
+- type: textarea
+  id: config
+  attributes:
+    label: Configuration
+    description: |
+      Paste the relevant parts of your `vector.toml` configuration.
```
Comment (binarylogic) [inline .github/ISSUE_TEMPLATE/bug.yml:36]:
```suggestion
      Paste the relevant parts of your Vector configuration file
```
We'll be migrating to YAML at some point in the future

Comment (jszwedko) [reply .github/ISSUE_TEMPLATE/bug.yml:36]: Ah, yes, good call 👍

Comment (jszwedko) [reply .github/ISSUE_TEMPLATE/bug.yml]: 81f8d461b

File: .github/ISSUE_TEMPLATE/feature.yml [hunk 2]
```diff
@@ -0,0 +1,95 @@
+name: Feature
+about: 🚀 Suggest a new feature.
+labels:
+- 'type: feature'
+
+---
+
+<!--
+-->
```
Comment (spencergilbert) [inline .github/ISSUE_TEMPLATE/feature.yml:9]: What's with these?

File: .github/ISSUE_TEMPLATE/feature.yml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+- type: markdown
+  attributes:
+    value: |
+      Thank you for opening 🚀 feature request!
+
+      For general questions about Vector usage, please see
+      https://discussions.vector.dev.
```
Comment (spencergilbert) [inline .github/ISSUE_TEMPLATE/feature.yml:16]: Do we also want to link to the GH discussions here?

**Review by spencergilbert — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/feature.yml (hunk 2 above)
Comment (jszwedko) [reply .github/ISSUE_TEMPLATE/feature.yml:9]: Oops! Those got left in by accident.

**Comment by jszwedko** [comment]
> Do you still want the enhancement tag?

Good question. I'll plan to delete it (after retagging enhancements). I don't know that there is much value in trying to differentiate features vs. enhancements. We've mostly been using it as an indication of level of effort which I think could be tracked in other ways.

**Pull request merged**